### PR TITLE
Rethink check_username

### PR DESCRIPTION
**Pull request by jirik**
Remove url_path authentication as side effect.

### CODE REVIEW

File: src/layman/error_list.py [hunk 1]
```diff
@@ -40,4 +40,5 @@
     37: (400, 'CSW exception.'),
     38: (400, 'Micka HTTP or connection error.'),
     39: (404, 'Metadata record does not exists.'),
+    40: (404, 'Username not yet reserved.'),
```
Comment (index-git) [inline src/layman/error_list.py:43]:
```suggestion
    40: (404, 'Username does not exist.'),
```

File: src/layman/util.py [hunk 2]
```diff
@@ -51,7 +51,13 @@ def to_safe_names(unsafe_names, type_name):
 def check_username_decorator(f):
     @wraps(f)
     def decorated_function(*args, **kwargs):
-        check_username(request.view_args['username'])
+        username = request.view_args['username']
+        if request.method == 'POST':
```
Comment (index-git) [inline src/layman/util.py:55]: Should there be tested, which POST method it is? I think, that only some POST methods can create user.

File: src/layman/error_list.py (hunk 1 above)
Comment (jirik) [reply src/layman/error_list.py:43]: Agree

File: src/layman/util.py (hunk 2 above)
Comment (jirik) [reply src/layman/util.py:55]: We could limit it only to POST Layers and POST Maps, but I think it’s not necessary. Creating user workspace on Post Layer Chunk does not break anything.
Anyway, if it makes sense to you to limit POST requests&check_username_decorator, I suggest to create separate issue for this and leave it for later.

Comment (index-git) [reply src/layman/util.py:55]: I am not sure what happened if I call POST with non existing username. I will check it and then either let it go or create separate issue.

**Pull request merged**